pex/cli/commands/lock.py

```diff
@@ -291,9 +292,35 @@ def _add_update_arguments(cls, update_parser):
             default=[],
             type=str,
             help=(
-                "Just attempt to update these projects in the lock, leaving all others unchanged. "
+                "Attempt to update these projects in the lock, leaving all others unchanged. "
                 "If the projects aren't already in the lock, attempt to add them as top-level "
-                "requirements leaving all others unchanged."
+                "requirements leaving all others unchanged. If a project is already in the lock "
+                "and is specified by a top-level requirement, the allowable updates will be "
+                "constrained by the original top-level requirement. In other words, for an "
+                "original top-level requirement of `requests>=2,<4` you could specify "
+                "`-p requests` to pick up new releases in the `>=2,<4` range specified by the "
+                "original top-level requirement or you could specify an overlapping range like "
+                "`-p 'requests<3.6,!==3.8.2'` to, for example, downgrade away from a newly "
+                "identified security vulnerability. If you specify a disjoint requirement like "
+                "`-p 'requests>=4'`, the operation will fail. If you really want to replace the "
+                "original top-level requirement and have this operation succeed, you can precede "
+                "the requirement with an `=` like so: `-p '=requests>=4'`. This option is mutually "
```

Comment:
I never really found any contingencies to force the choice between a new option, say `-r` / `--replace-project` and this `=`-prefix syntax on the existing `-p` / `--project` option like I did with `-d` / `--delete`. I'm ambivalent; so if you have some good reason this new replace should be split into its own flag, please speak up.

Reply:
@kaos helped answer this here: https://github.com/pantsbuild/pex/issues/2332#issuecomment-1900318716

I'll fix by splitting out `-R` / `--replace-project` (`-r` is / will be taken by adding support for bare and `-r` requirements in a forthcoming PR).